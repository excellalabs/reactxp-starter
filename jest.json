{
  "preset": "react-native",
  "transform": {
    "\\.(ts|tsx)$": "ts-jest/preprocessor.js",
    "\\.(js|jsx)$": "babel-jest"
  },
  "testRegex": "(src|storybook)/.*(/__tests__/.*|\\.(spec))\\.(ts|tsx|js)$",
  "moduleFileExtensions": [
    "ts",
    "tsx",
    "js",
    "json"
  ],
  "globals": {
    "ts-jest": {
      "tsConfigFile": "tsconfig.json"
    }
  },
  "moduleNameMapper": {
    "^commonStyles$": "<rootDir>/src/commonStyles/index.ts",
    "^commonStyles/(.*)$": "<rootDir>/src/commonStyles/$1.ts",
    "^components/(.*)$": "<rootDir>/src/components/$1/index.ts",
    "^containers/(.*)$": "<rootDir>/src/containers/$1/index.ts",
    "^helpers/(.*)$": "<rootDir>/src/helpers/$1.ts",
    "^queries/(.*)$": "<rootDir>/src/queries/$1.ts",
    "^redux-store$": "<rootDir>/src/redux-store/index.ts",
    "^actions/(.*)$": "<rootDir>/src/redux-store/actions/$1.ts",
    "^thunks/(.*)$": "<rootDir>/src/redux-store/thunks/$1.ts",
    "^routes/(.*)$": "<rootDir>/src/routes/$1/index.ts",
    "^services/(.*)$": "<rootDir>/src/services/$1.ts"
  },
  "collectCoverage": true,
  "collectCoverageFrom": [
    "src/**/*.{ts,tsx}",
    "!src/index.tsx",
    "!src/**/*.d.ts",
    "!src/redux-store/index.ts",
    "!src/redux-store/rootReducer.ts",
    "!src/redux-store/store.ts",
    "!src/routes/**/*"
  ],
  "coverageThreshold": {
    "global": {
      "branches": 20,
      "functions": 20,
      "lines": 20,
      "statements": 20
    }
  }
}