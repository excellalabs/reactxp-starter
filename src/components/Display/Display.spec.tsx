
import 'jest'
import 'react-native'
import * as React from 'react'
import OriginalDisplay from '.'

type DisplayType = typeof OriginalDisplay

// Note: test renderer must be required after react-native.
import * as renderer from 'react-test-renderer'

const importModule = (): DisplayType => {
  const disp = require('.').default
  return disp
}

const mockedComponents = [
]

const defaultMocks = [
]

const allMocks = mockedComponents.concat(defaultMocks)

describe('<Display />', () => {
  beforeEach(() => {
    jest.resetModules()
    mockedComponents.forEach(component => {
      jest.setMock(component, component)
    })
    defaultMocks.forEach(component => {
      jest.setMock(component, jest.genMockFromModule(component))
    })
  })

  afterEach(() => {
    allMocks.forEach(component => {
      jest.unmock(component)
    })
  })

  describe('snapshots', () => {
    it('renders correctly', () => {
      const Display = importModule()
      const tree = renderer.create(
        <Display />
      )

      expect(tree).toMatchSnapshot()
    })
  })
})
