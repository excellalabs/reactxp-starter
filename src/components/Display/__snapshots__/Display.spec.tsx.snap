// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Display /> snapshots renders correctly 1`] = `
<RX.View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <RX.Text
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    Your App
  </RX.Text>
  <RX.Text
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  />
</RX.View>
`;
